pkg/scalers/apache_kafka_scaler.go

```diff
@@ -120,84 +120,148 @@ func NewApacheKafkaScaler(ctx context.Context, config *scalersconfig.ScalerConfi
 
 func parseApacheKafkaAuthParams(config *scalersconfig.ScalerConfig, meta *apacheKafkaMetadata) error {
 	meta.enableTLS = false
-	enableTLS := false
-	if val, ok := config.TriggerMetadata["tls"]; ok {
-		switch val {
-		case stringEnable:
-			enableTLS = true
-		case stringDisable:
-			enableTLS = false
-		default:
-			return fmt.Errorf("error incorrect TLS value given, got %s", val)
-		}
-	}
-
-	if val, ok := config.AuthParams["tls"]; ok {
-		val = strings.TrimSpace(val)
-		if enableTLS {
-			return errors.New("unable to set `tls` in both ScaledObject and TriggerAuthentication together")
-		}
-		switch val {
-		case stringEnable:
-			enableTLS = true
-		case stringDisable:
-			enableTLS = false
-		default:
-			return fmt.Errorf("error incorrect TLS value given, got %s", val)
-		}
+	var enableTLS bool
+	tlsString, err := getParameterFromConfigV2(
+		config,
+		"tls",
+		reflect.TypeOf(""),
+		UseMetadata(true),
+		UseAuthentication(true),
+		UseResolvedEnv(false),
+		IsOptional(true),
+		WithDefaultVal("disable"),
```

Comment:
nit: iirc, the default value for optional arg is false, so the `UseResolvedEnv(false)` could be omitted? and for brevity, can we omit the arguments?

```suggestion
	tlsString, err := getParameterFromConfigV2(
		config,
		"tls",
		reflect.TypeOf(""),
		UseMetadata(),
		UseAuthentication(),
		IsOptional(),
		WithDefaultVal("disable"),
```

Reply:
I think eleminating option function when it is not `true` make sense as we already initiate the default value in this 
https://github.com/kedacore/keda/blob/ef464594c311818fb400cc8d69a8d6e9721da21b/pkg/scalers/scaler.go#L243

As fo having the default value, I personally does not enjoy that as Golang does not support default arg value out of the box :laughing: , although we can still make some changes to adopt this, for example 
Before 
```golang
func IsOptional(optional bool) Option {
	return func(opt *configOptions) {
		opt.isOptional = optional
	}
}
```
After
```golang
func IsOptional(optionals ...bool) Option {
    optional := true // Default value
    if len(optionals) > 0 {
        optional = optionals[0]
    }
    return func(opt *configOptions) {
        opt.isOptional = optional
    }
}
```
Let me know if you still want to go through with the second point on default args value